### PR TITLE
update UserGuide for filter command

### PR DESCRIPTION
**Pull request by Acerizm**
1. Description of filter command added after `Editing the data file` description
2. Added images in `docs/image/filter` directory for filter commands
3. Updated `Command Summary` to include filter command

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -167,9 +167,19 @@ AddressBook data are saved as a JSON file `[JAR file location]/data/addressbook.
...
+Filters the entries that matches the tag provided. The tag provided can be in uppercase,lowercase or mixed.
+
+Format: `filter [TAG]`
+
+Examples:
+
+* `filter girlfriend` returns `No contacts matches the tag provided!`
+
+![filterGirlfriend](images/filter/filterGirlfriendResult.png)
+* `filter family` returns `Alex Yeoh`
```
Comment (joellow88) [inline docs/UserGuide.md:181]: The filterFamilyResult.png image should have Alex Yeoh tagged as family instead of friend

**Review by joellow88 — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 1 above)
Comment (Acerizm) [reply docs/UserGuide.md:181]: Noted! Thanks for pointing it out

**Comment by Acerizm** [comment]
@joellow88 filterFamilyResult image has been modified to show family tag instead. Help me take a look and see if it's correct!

**Review by joellow88 — APPROVED** [review]
Thanks for changing the image, LGTM!

**Pull request merged**